pod.go

```diff
@@ -389,6 +389,11 @@ func (p *Pod) Annotations(key string) (string, error) {
 	return value, nil
 }
 
+// SetOrAddAnnotation sets or adds an annotation
+func (p *Pod) SetOrAddAnnotation(key, value string) {
```

Comment:
I would prefer `SetAnnotation()` instead of `SetOrAddAnnotation`, because the way you have implemented this, there is no way to know if it was added or replaced. But keep the comment like this because it explains what is done.